### PR TITLE
CustomerAgent.java

### PR DESCRIPTION
**Pull request by widyaaulia**
Customer Agent takes input from json file, send order to all sellers, receive product price from sellers, pick the cheapest sellers, and send order to cheapest sellers

### CODE REVIEW

**Comment by janhavi19** [comment]
The error is caused because the bulid gradle does not contain the Json dependency

**Comment by argenos** [comment]
Please add the gradle dependency to your pull request! 

What is the purpose of the Data class that you didn't include?

**Comment by argenos** [comment]
Why is that not part of your customer?

**Comment by widyaaulia** [comment]
yes. that would be my mistakes as I only pulled them from upstream without checking it. Regarding the utils, I don't have any reason for that. I re-pull requested just now after fixing the issues.

**Comment by janhavi19** [comment]
We made a separate class for code readability.

**Comment by argenos** [comment]
The missing functionality outweighs the code readability. Try to group your code in logical blocks and keep in mind the naming of your classes. 

In this particular case, I believe the functionality to check for prices and choose a bakery belongs to your customer and not to that “Data” class.

**Comment by janhavi19** [comment]
Could you please tell us what functionalities are missing

**Comment by argenos** [comment]
The Travis build won’t pass because some of your customer functionalities are taking place in your “Data” class, which you didn’t include. Other teams won’t be able to use it this way. While you could just add that file to your pull request, that is not a great way to organize your code. 

From your response above, the Data class performing the checks for the lowest price. Either the naming of the class is incorrect, or the functions belong on your customer class. Depending on this, you should update your pull request.

**Comment by widyaaulia** [comment]
I checked, but it has passed the Travis build after I repullrequested it. However, I put the class Data in utils folder. Is it preferable to put it inside the class CustomerAgent?

**Comment by argenos** [comment]
Yes, that would be better.

File: src/main/java/org/yourteamname/utils/Data.java [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+	Orders are randomly generated. They are saved in JSON format and includes ID and ordered
+	product list. 
+	16 Nov 2018: only one order in orders. Customer ID is randomly generated to make sure 
+	there are no duplication. Number of product ordered is also randomly generated. 
+	*/ 
+	public List<JSONObject> makeOrder(List<String> product_types) {
+		List<JSONObject> orders = new ArrayList<>();
+		JSONObject order = new JSONObject();
+				
+		String customer_id = generateRandomID(10);
```
Comment (argenos) [inline src/main/java/org/yourteamname/utils/Data.java]: Customers already have a guid, you should be using the one from the scenario files.

**Comment by argenos** [comment]
In general, since these are the customers @HBRS-MAAS will need to use, they are the ones that need to test and give feedback.

**Comment by argenos** [comment]
To clarify this a bit, testing of the customers and reporting issues is part of Wednesday's lab class (but not the only activity).

File: src/main/java/org/yourteamname/utils/Data.java (hunk 1 above)
Comment (widyaaulia) [reply src/main/java/org/yourteamname/utils/Data.java]: The function is no longer used by Customer Agent. I just haven't deleted it yet. It was from last week assignment where I haven't used the scenario files yet. 

I will include all customer function to CustomerAgent.java as you suggested.

**Comment by argenos** [comment]
@HBRS-MAAS so far no one has provided feedback to this pull request, if this is in fact approved by the class, I will merge this Thursday evening.

**Pull request merged**